website/docs/foundations/typography/partials/guidelines/guidelines.md

```diff
@@ -0,0 +1,107 @@
+
+### Body
+
+Body styles should be used for all general page content. Commonly placed within components (like buttons) and blocks of text, this style supports different font weights and sizes to establish a clear visual hierarchy. Body 200 is used as the default body content size as the majority of HashiCorp products are data dense. Body 100 is often used as secondary text for hierarchical purposes while Body 300 is used for bigger UI elements like the `large` variant of the `Button` component.
+
+Body text styles are available in the following weights and sizes:
+
+![Body 300 at 16px font size, with semibold, medium and regular font weights. Body 200 at 14px font size, with semibold, medium and regular font weights. Body 100 at 13px font size, with semibold, medium and regular font weights.](/assets/foundations/typography/typography-body-options.png)
+
+### Code
+
+Code styles are used exclusively for text representing code in the UI (like the [Code Block component](/components/code-block)). 
+
+Code text styles come in the following weights and sizes:
+
+![Code 300 at 16px font size, with bold and regular font weights. Code 200 at 14px font size, with bold and regular font weights. Bode 100 at 13px font size, with bold and regular font weights.](/assets/foundations/typography/typography-code-options.png)
+
+## How typography is used
+
+### Body content
+
+Body 200 is the default font size for body content. Scaling up or down is used as a means to create visual hierarchy in page UI, or component usage. As an example, Helio’s `Button`, `medium` (default) variant is sized at Body 200 while the `large` variant is Body 300.
+
+![Two buttons, one medium and one large](/assets/foundations/typography/typography-button-example.png)
+
+Similarly, form fields’ labels and input texts are Body 200, while any other secondary information, like helper text or error messages are sized at Body 100. This ensures that from a hierarchy perspective, the primary information is more prominent with the larger size of the two.
+
+![A form field set with label, helper text an input field errored out along with an error message.](/assets/foundations/typography/typography-form-example.png)
+
+For more details about form patterns, read the [form patterns](/patterns/form-patterns) documentation for additional details.
+
+### Semantic heading or not?
+
+Even though Helio’s Text Component does not equate a specific style (Body, Display, or Code) with a specific HTML semantic structure (like `<p/>`, `<h1/>` or `<span>`), this is up to the consumer to define the logical order in which the text is being displayed as. To ensure accessible page content, the scaling of the Display styles often are associated with the scaling of the HTML headings such as:
```

Comment:
```suggestion
As Helio’s Text component does not equate to a specific style (Body, Display, or Code) or a semantic HTML element (like `<h1/>`, `<h2/>`, etc.), it is up to the consumer to define the tag with the appropriate sequential order for the heading text. To ensure accessible page content, the scaling of the Display styles are often associated with the scaling of the HTML headings such as:
```

Reply:
This section is related to appropriate use of header styles & HTML markup while the text was more general so seemed confusing to me. (So I made the above suggested edits)

Reply:
[suggestion] I would also remove the reference to Helios (redundant)
```suggestion
Even though the [Text](/components/text) component does not equate a specific style (Body, Display, or Code) with a specific HTML semantic structure (like `<p>`, `<h1>` or `<span>`), this is up to the consumer to define the logical order in which the text is being displayed as. To ensure accessible page content, the scaling of the Display styles often are associated with the scaling of the HTML headings such as:
```

Reply:
I would even be a little more forceful with this: 
```suggestion
The Helio’s Text Component does not equate a specific style (Body, Display, or Code) with a specific HTML semantic structure (like `<p/>`, `<h1/>` or `<span>`), it is up to the consumer to define the logical semantic element for the each piece of text. To ensure accessible page content, the scaling of the Display styles often are associated with the scaling of the HTML headings such as:
```